demos/imagebuilder-windowsbaseline/main.bicep

```diff
@@ -0,0 +1,212 @@
+@description('The base URI where artifacts required by this template are located including a trailing \'/\'')
+param _artifactsLocation string = deployment().properties.templateLink.uri
+
+@description('The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated. Use the defaultValue if the staging location is not secured.')
+@secure()
+param _artifactsLocationSasToken string = ''
+
+@description('The Azure region where resources in the template should be deployed.')
+param location string = resourceGroup().location
+
+@description('The name of the customizer script which will be executed during image build.')
+param customizerScriptName string = 'scripts/runScript.ps1'
+
+@description('Name of the user-assigned managed identity used by Azure Image Builder template, and for triggering the Azure Image Builder build at the end of the deployment')
+param identityName string = substring('ImageGallery_${guid(resourceGroup().id)}', 0, 21)
+
+@description('Permissions to allow for the user-assigned managed identity.')
+param roleDefinitionName string = guid(resourceGroup().id)
+
+@description('Name of the role to grant for the user-assigned managed identity.')
+param roleAssignmentName string = guid(resourceGroup().id)
```

Comment:
this needs to incorporate the principal, role and scope - and you'll want to make it a variable so users can't change it (and fail)

Reply:
ack